File: src/codes/Users_Manual/gain/group_0.cpp
//~#include<optional>
//~#include<autd3.hpp>
//~int main() {
//~using namespace autd3;
//~const auto x = 0.0;
//~const auto y = 0.0;
//~const auto z = 0.0;
gain::Group(
    [](const auto& dev) {
      return [](const auto& tr) -> std::optional<const char*> {
        if (tr.idx() <= 100) return "null";
        return "focus";
      };
    },
    std::unordered_map<const char*, std::shared_ptr<Gain>>{
        {"focus", std::make_shared<Focus>(Point3(x, y, z), FocusOption{})},
        {"null", std::make_shared<Null>()}});
//~return 0; }

File: src/codes/Users_Manual/gain/group_0.rs
use autd3::prelude::*;
# use std::collections::HashMap;

# fn main() -> Result<(), Box<dyn std::error::Error>> {
# let x = 0.;
# let y = 0.;
# let z = 0.;
# let _ =
GainGroup::new(
    |_dev| {
        |tr| match tr.idx() {
            0..=100 => Some("null"),
            _ => Some("focus"),
        }
    },
    HashMap::from([
        ("null", BoxedGain::new(Null {})),
        (
            "focus",
            BoxedGain::new(Focus {
                pos: Point3::new(x, y, z),
                option: Default::default(),
            }),
        ),
    ]),
);
# Ok(())
# }